### PR TITLE
OpenMPTarget: Update docker clang build.

### PR DESCRIPTION
**Pull request by rgayatri23**
This is a draft PR to update dockerfile for OpenMPTarget.

### CODE REVIEW

File: scripts/docker/Dockerfile.openmptarget [hunk 1]
```diff
@@ -55,7 +55,6 @@ RUN LLVM_URL=https://github.com/llvm/llvm-project/archive &&\
       -DCMAKE_CXX_COMPILER=g++ \
       -DLLVM_ENABLE_PROJECTS="clang" \
       -DLLVM_ENABLE_RUNTIMES="openmp" \
-      -DCLANG_OPENMP_NVPTX_DEFAULT_ARCH=sm_70 \
       -DLIBOMPTARGET_NVPTX_COMPUTE_CAPABILITIES=70 \
```
Comment (masterleinad) [inline scripts/docker/Dockerfile.openmptarget:58]: Don't you want to use
````
-DLIBOMPTARGET_DEVICE_ARCHITECTURES=sm_70
```
instead?

Comment (rgayatri23) [reply scripts/docker/Dockerfile.openmptarget:58]: No, this is sufficient `-DLIBOMPTARGET_NVPTX_COMPUTE_CAPABILITIES=70 ` . The default arch option is getting ignored actually.

**Review by masterleinad — APPROVED** [review]
Fine with me.

File: scripts/docker/Dockerfile.openmptarget (hunk 1 above)
Comment (dalg24) [reply scripts/docker/Dockerfile.openmptarget:58]: Would you post a link to a reference for the build options doc?

Comment (rgayatri23) [reply scripts/docker/Dockerfile.openmptarget:58]: This is the best documentation I found [wiki](https://hpc-wiki.info/hpc/Building_LLVM/Clang_with_OpenMP_Offloading_to_NVIDIA_GPUs) .
While it still says use the `DEFAULT_ARCH`, cmake actually ignores it. So I guess the documentation is not up-to date

Comment (masterleinad) [reply scripts/docker/Dockerfile.openmptarget:58]: I was looking at https://openmp.llvm.org/SupportAndFAQ.html#q-how-to-build-an-openmp-nvidia-offload-capable-compiler.

**Review by dalg24 — DISMISSED** [review]

File: scripts/docker/Dockerfile.openmptarget (hunk 1 above)
Comment (dalg24) [reply scripts/docker/Dockerfile.openmptarget:58]: Wait the doc says to use `-DLIBOMPTARGET_DEVICE_ARCHITECTURES=sm_<xy>`

Comment (dalg24) [reply scripts/docker/Dockerfile.openmptarget:58]: We should be using `LIBOMPTARGET_DEVICE_ARCHITECTURES` (confirmed on the llvm/openmp discord channel)

Comment (dalg24) [inline scripts/docker/Dockerfile.openmptarget]:
```suggestion
      -DLIBOMPTARGET_DEVICE_ARCHITECTURES=sm_70 \
```

**Pull request merged**